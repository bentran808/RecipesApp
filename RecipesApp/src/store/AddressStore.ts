import { toJS } from 'mobx';
import { types, destroy, SnapshotOut, getRoot, applySnapshot } from 'mobx-state-tree';
import { RootStore } from 'store/store';

const AddressEntry = types
  .model('AddressEntry', {
    id: types.optional(types.identifierNumber, 0),
    type: types.optional(types.string, ''),
    address: types.optional(types.string, '')
  })
  .actions((self) => ({
    remove() {
      getRoot<typeof RootStore>(self).address.removeItem(self);
    },
    edit(type: string, address: string) {
      self.type = type;
      self.address = address;
    }
  }));

export type AddressModel = SnapshotOut<typeof AddressEntry>;

const AddressStore = types
  .model('', {
    items: types.array(AddressEntry),
    itemUsing: types.optional(AddressEntry, {})
  })
  .volatile(() => ({
    address: ''
  }))
  .actions((self) => ({
    setAddress(address: string) {
      self.address = address;
    },
    addNewItem(type: string, address: string) {
      const id = self.items.reduce((maxId, item) => Math.max(item.id, maxId), -1) + 1;
      self.items.push({ id, type, address });
    },
    removeItem(item: AddressModel) {
      destroy(item);
    },
    existAddress(address?: AddressModel) {
      return self.items.find((item) => item.id === address?.id);
    },
    setItemUsing(item: AddressModel) {
      applySnapshot(self, {
        ...self,
        itemUsing: item
      });
    }
  }))
  .views((self) => ({
    get addresses() {
      return toJS(self.items);
    }
  }));

export default AddressStore;
