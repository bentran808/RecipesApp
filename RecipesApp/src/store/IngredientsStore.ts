import { recipesApi } from 'api';
import { toJS } from 'mobx';
import { flow, SnapshotOut, types } from 'mobx-state-tree';

export const IngredientsEntry = types.model('IngredientsEntry', {
  id: types.number,
  name: types.string,
  photo_url: types.string
});

export type IngredientModel = SnapshotOut<typeof IngredientsEntry>;

const IngredientsStore = types
  .model('IngredientsStore', {
    lists: types.optional(types.array(IngredientsEntry), []),
    state: types.optional(types.enumeration('State', ['pending', 'done', 'error']), 'done')
  })
  .views((self) => ({
    get ingredients() {
      return self.lists;
    },
    get ingredientsJS() {
      return toJS(this.ingredients);
    }
  }))
  .actions((self) => ({
    fetchIngredients: flow(function* fetchIngredients() {
      self.state = 'pending';
      try {
        const response = yield recipesApi.fetchIngredientsRequest();
        self.lists = response.data;
        self.state = 'done';
      } catch (error) {
        self.state = 'error';
      }
    })
  }));

export default IngredientsStore;
