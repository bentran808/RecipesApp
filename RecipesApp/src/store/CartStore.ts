import { toJS } from 'mobx';
import { applySnapshot, destroy, getRoot, SnapshotOut, types } from 'mobx-state-tree';
import { RecipeModel, RecipesEntry } from 'store/RecipesStore';
import { RootStore } from 'store/store';

export const CartEntry = types
  .model('CartEntry', {
    item: RecipesEntry,
    quantity: types.optional(types.number, 1),
    inCart: types.optional(types.boolean, true)
  })
  .actions((self) => ({
    remove() {
      getRoot<typeof RootStore>(self).cart.removeItem(self);
    },
    increase(quantity?: number) {
      self.quantity = quantity ? self.quantity + quantity : ++self.quantity;
    },
    decrease() {
      self.quantity === 1 ? this.remove() : --self.quantity;
    },
    toggle() {
      self.inCart = !self.inCart;
    }
  }));

const BillEntry = types.model('BillEntry', {
  type: types.optional(types.string, ''),
  name: types.optional(types.string, ''),
  code: types.optional(types.string, ''),
  price: types.optional(types.number, 0)
});

const OrderedEntry = types.model('OrderedEntry', {
  items: types.array(
    types.model({
      name: types.optional(types.string, ''),
      quantity: types.optional(types.number, 0)
    })
  ),
  total: types.number,
  createdAt: types.string
});

export type CartModel = SnapshotOut<typeof CartEntry>;
export type BillModel = SnapshotOut<typeof BillEntry>;
export type OrderedModel = SnapshotOut<typeof OrderedEntry>;

const CartStore = types
  .model('CartStore', {
    items: types.array(CartEntry),
    billItems: types.array(BillEntry),
    orderedItems: types.array(OrderedEntry)
  })
  .views((self) => ({
    get recipes() {
      return toJS(self.items);
    },
    get recipesInCart() {
      return this.recipes.filter((item) => item.inCart);
    },
    get inCartCount() {
      return this.recipesInCart.length;
    },
    get total() {
      return this.recipesInCart.reduce((sum, item) => sum + item.quantity * 20, 0);
    },
    get billDetails() {
      return toJS(self.billItems);
    },
    get toPay() {
      return this.billDetails.reduce(
        (sum, item) => (item.type === 'discount' ? sum - item.price : sum + item.price),
        0
      );
    },
    get orderedList() {
      return toJS(self.orderedItems);
    }
  }))
  .volatile(() => ({
    discountInput: '',
    invalidCoupon: false
  }))
  .actions((self) => ({
    addToCart(item: RecipeModel, quantity?: number) {
      const itemInCart = this.hasInCart(item);
      itemInCart ? itemInCart.increase(quantity) : self.items.push({ item, quantity });
    },
    removeItem(item: CartModel) {
      destroy(item);
    },
    hasInCart(recipe: RecipeModel) {
      return self.items.find((item) => item.item.id === recipe.id);
    },
    setDiscountInput(text: string) {
      if (!text) {
        self.invalidCoupon = false;
      }
      self.discountInput = text;
    },
    applyCoupon(item: BillModel) {
      const existItem = self.billItems.findIndex((bill) => bill.type === 'discount');
      if (existItem > -1) {
        self.billItems.splice(existItem, 1, item);
      } else {
        self.billItems.push(item);
      }
    },
    payment(order: OrderedModel) {
      self.orderedItems.unshift(order);
      applySnapshot(self, {
        ...self,
        items: [],
        billItems: []
      });
    },
    isInvalidCoupon(status: boolean) {
      self.invalidCoupon = status;
    }
  }));

export default CartStore;
