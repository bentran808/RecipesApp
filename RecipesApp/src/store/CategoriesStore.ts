import { recipesApi } from 'api';
import { toJS } from 'mobx';
import { flow, SnapshotOut, types } from 'mobx-state-tree';
import { transformCategories } from 'utils';
import { RecipesEntry } from './RecipesStore';

export const CategoriesEntry = types.model('CategoriesEntry', {
  id: types.number,
  name: types.string,
  photo_url: types.string,
  recipes: types.maybe(types.array(RecipesEntry))
});

export type CategoryModel = SnapshotOut<typeof CategoriesEntry>;

const CategoriesStore = types
  .model('CategoriesStore', {
    lists: types.optional(types.array(CategoriesEntry), []),
    state: types.optional(types.enumeration('State', ['pending', 'done', 'error']), 'done')
  })
  .views((self) => ({
    get categories() {
      return self.lists;
    },
    get categoriesJS() {
      return transformCategories(toJS(this.categories));
    }
  }))
  .actions((self) => ({
    fetchCategories: flow(function* fetchCategories() {
      self.state = 'pending';
      try {
        const response = yield recipesApi.fetchCategoriesRequest();
        self.lists = response.data;
        self.state = 'done';
      } catch (error) {
        console.log(error);
        self.state = 'error';
      }
    })
  }));

export default CategoriesStore;
