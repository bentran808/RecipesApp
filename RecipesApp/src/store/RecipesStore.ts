import { recipesApi } from 'api';
import { toJS } from 'mobx';
import { applySnapshot, flow, SnapshotOut, types } from 'mobx-state-tree';
import { CategoryModel } from 'store/CategoriesStore';

export const RecipesEntry = types.model('RecipesEntry', {
  id: types.number,
  categoryId: types.number,
  title: types.string,
  photo_url: types.string,
  photosArray: types.array(types.string),
  time: types.string,
  ingredients: types.array(types.array(types.union(types.number, types.string))),
  description: types.string,
  category: types.maybe(
    types.model({
      id: types.number,
      name: types.string,
      photo_url: types.string
    })
  )
});

export type RecipeModel = SnapshotOut<typeof RecipesEntry>;

const RecipesStore = types
  .model('RecipesStore', {
    items: types.optional(types.array(RecipesEntry), []),
    state: types.optional(types.enumeration('State', ['pending', 'done', 'error']), 'done'),
    currentPage: types.optional(types.number, 1),
    isEndList: types.optional(types.boolean, false)
  })
  .views((self) => ({
    get recipesJS() {
      return toJS(self.items);
    }
  }))
  .volatile(() => ({
    keyword: ''
  }))
  .actions((self) => ({
    fetchRecipes: flow(function* fetchRecipes(page?: number) {
      if (!self.isEndList) {
        self.state = 'pending';
        try {
          const response = yield recipesApi.fetchRecipesRequest(page);
          if (response.data.length) {
            self.currentPage++;
            applySnapshot(self, {
              ...self,
              items: !page || page === 1 ? response.data : [...self.items, ...response.data]
            });
          } else {
            self.isEndList = true;
          }
          self.state = 'done';
        } catch (error) {
          self.state = 'error';
        }
      }
    }),
    fetchRecipesByCategoryId: flow(function* fetchRecipesByCategoryId(id: number) {
      self.state = 'pending';
      if (self.isEndList) {
        self.currentPage = 1;
        self.isEndList = false;
      }
      try {
        const response = yield recipesApi.fetchRecipesByCategoryIdRequest(id);
        self.items = response.data;
        self.state = 'done';
      } catch (error) {
        self.state = 'error';
      }
    }),
    searchRecipeName: flow(function* searchRecipeName(text: string) {
      self.state = 'pending';
      if (self.isEndList) {
        self.currentPage = 1;
        self.isEndList = false;
      }
      try {
        const response = yield recipesApi.searchByRecipeNameRequest(text);

        applySnapshot(self, {
          ...self,
          items: [...self.items, ...response.data]
        });
        self.state = 'done';
      } catch (error) {
        self.state = 'error';
      }
    }),
    searchCategoryName: flow(function* searchCategoryName(text: string) {
      self.state = 'pending';
      if (self.isEndList) {
        self.currentPage = 1;
        self.isEndList = false;
      }
      try {
        const response = yield recipesApi.searchByCategoryNameRequest(text);
        const recipesOfCategory = (response.data || [])
          .map((category: CategoryModel) =>
            (category.recipes || []).map((recipe) => ({
              ...recipe,
              category: {
                id: category.id,
                name: category.name,
                photo_url: category.photo_url
              }
            }))
          )
          .flat(1);

        applySnapshot(self, {
          ...self,
          items: [...self.items, ...recipesOfCategory]
        });
        self.state = 'done';
      } catch (error) {
        self.state = 'error';
      }
    }),
    setKeyword(text: string) {
      self.keyword = text;
    },
    setRefreshData() {
      self.currentPage = 1;
      self.isEndList = false;
    },
    setEmptyRecipes() {
      applySnapshot(self, {
        ...self,
        items: []
      });
    }
  }));

export default RecipesStore;
