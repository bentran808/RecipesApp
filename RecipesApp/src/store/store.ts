import { Instance, types } from 'mobx-state-tree';
import RecipesStore from 'store/RecipesStore';
import CategoriesStore from './CategoriesStore';

const RootStore = types.model('RootStore', {
  categories: CategoriesStore,
  recipes: RecipesStore
});

const store = RootStore.create({
    categories: {},
    recipes: {}
});

export type RootStoreModel = Instance<typeof RootStore>;

export default store;
