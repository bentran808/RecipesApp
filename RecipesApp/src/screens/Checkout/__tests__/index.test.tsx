import { fireEvent, render } from '@testing-library/react-native';
import { StoreProvider } from 'context';
import { recipe } from 'mocks';
import React from 'react';
import renderer from 'react-test-renderer';
import store, { RootStore } from 'store/store';
import CheckoutScreen from '..';

describe('Checkout Screen Component', () => {
  const appStore = RootStore.create({
    categories: {},
    recipes: {},
    ingredients: {},
    cart: {
      items: [
        {
          item: recipe,
          quantity: 1,
          inCart: true
        }
      ],
      billItems: [
        {
          type: 'total',
          name: 'Item Total',
          code: 'TOTAL',
          price: 20
        },
        {
          type: 'fee',
          name: 'Delivery Fee',
          code: 'DELIVER',
          price: 5
        },
        {
          type: 'discount',
          name: 'Free Ship',
          code: 'FREESHIP',
          price: 5
        }
      ]
    },
    address: {
      itemUsing: {
        id: 0,
        type: 'Work',
        address: '604 NT'
      }
    }
  });
  const navigation: any = {
    navigate: jest.fn()
  };

  test('should render correctly', () => {
    const tree = renderer
      .create(
        <StoreProvider value={appStore}>
          <CheckoutScreen navigation={navigation} />
        </StoreProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly with error message', () => {
    appStore.cart.isInvalidCoupon(true);
    const tree = renderer
      .create(
        <StoreProvider value={appStore}>
          <CheckoutScreen navigation={navigation} />
        </StoreProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function handlePressApply with invalid coupon', () => {
    store.cart.isInvalidCoupon = jest.fn();
    const { getByPlaceholderText, getByTestId } = render(
      <StoreProvider value={store}>
        <CheckoutScreen navigation={navigation} />
      </StoreProvider>
    );

    const input = getByPlaceholderText('Enter discount code');
    fireEvent.changeText(input, 'a');

    const button = getByTestId('applyBtn');
    fireEvent.press(button);

    expect(store.cart.isInvalidCoupon).toHaveBeenCalled();
  });

  test('should call function handlePressApply with coupon freeship', () => {
    store.cart.applyCoupon = jest.fn();
    const { getByPlaceholderText, getByTestId } = render(
      <StoreProvider value={store}>
        <CheckoutScreen navigation={navigation} />
      </StoreProvider>
    );

    const input = getByPlaceholderText('Enter discount code');
    fireEvent.changeText(input, 'freeship');

    const button = getByTestId('applyBtn');
    fireEvent.press(button);

    expect(store.cart.applyCoupon).toHaveBeenCalledWith({
      code: 'FREESHIP',
      name: 'Free Ship',
      price: 5,
      type: 'discount'
    });
  });

  test('should call function handlePressApply with coupon get10off', () => {
    store.cart.applyCoupon = jest.fn();
    const { getByPlaceholderText, getByTestId } = render(
      <StoreProvider value={store}>
        <CheckoutScreen navigation={navigation} />
      </StoreProvider>
    );

    const input = getByPlaceholderText('Enter discount code');
    fireEvent.changeText(input, 'get10off');

    const button = getByTestId('applyBtn');
    fireEvent.press(button);

    expect(store.cart.applyCoupon).toHaveBeenCalledWith({
      code: 'GET10OFF',
      name: 'Offer 10% OFF',
      price: 0,
      type: 'discount'
    });
  });

  test('should call function handlePressApply with coupon get20off', () => {
    store.cart.applyCoupon = jest.fn();
    const { getByPlaceholderText, getByTestId } = render(
      <StoreProvider value={store}>
        <CheckoutScreen navigation={navigation} />
      </StoreProvider>
    );

    const input = getByPlaceholderText('Enter discount code');
    fireEvent.changeText(input, 'get20off');

    const button = getByTestId('applyBtn');
    fireEvent.press(button);

    expect(store.cart.applyCoupon).toHaveBeenCalledWith({
      code: 'GET20OFF',
      name: 'Offer 20% OFF',
      price: 0,
      type: 'discount'
    });
  });

  test('should call function handlePressPayment', () => {
    const { getByTestId } = render(
      <StoreProvider value={appStore}>
        <CheckoutScreen navigation={navigation} />
      </StoreProvider>
    );
    const button = getByTestId('paymentBtn');
    fireEvent.press(button);

    expect(navigation.navigate).toHaveBeenCalled();
  });
});
