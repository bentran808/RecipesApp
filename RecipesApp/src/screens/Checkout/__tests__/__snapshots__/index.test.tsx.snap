// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Screen Component should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        data={
          Array [
            Object {
              "inCart": true,
              "item": Object {
                "category": Object {
                  "id": 0,
                  "name": "Cookies",
                  "photo_url": "https://www.telegraph.co.uk/content/dam/Travel/2019/January/france-food.jpg?imw'id'th=1400",
                },
                "categoryId": 0,
                "description": "-- Start with cleaned and peeled russet potatoes that you have cut into 3/8-inch match sticks. Place in bowl of very cold water: keep rinsing and changing the water until the water is clear; drain thoroughly and dry with paper towels or a clean lint-free kitchen towel.

 -- Meanwhile, you preheat your hot oil to 350 degrees F. Place prepared taters in oil and cook about 5 minutes. They will have that blond-tone color to them. 

 -- Note: Once you add cold potatoes to the hot oil, the temperature of your oil is going to drop - you want it to be somewhere between 330 - 325 degrees F. 

 -- Remove from oil; drain and cool. Now - either refrigerate until ready to finish cooking, or cool completely and freeze up to 3 months. To freeze properly - place completely cooled fries in single layer on tray and place in freezer until frozen. Then bag them.

 -- To finish cooking - preheat your oil to 400* F. Add your cold fries (which will drop the oil temp - which is fine because you want it near the 375 degrees F. temp) and cook a few minutes until done. Lightly salt them and shake well so that the salt distributes well and they are not salty.",
                "id": 0,
                "ingredients": Array [
                  Array [
                    0,
                    "200ml",
                  ],
                  Array [
                    1,
                    "5g",
                  ],
                  Array [
                    2,
                    "300g",
                  ],
                ],
                "photo_url": "https://www.texanerin.com/content/uploads/2019/06/nobake-chocolate-cookies-1-650x975.jpg",
                "photosArray": Array [
                  "https://www.texanerin.com/content/uploads/2019/06/nobake-chocolate-cookies-1-650x975.jpg",
                  "https://namelymarly.com/wp-content/uploads/2018/04/20180415_Beet_Lasagna_10.jpg",
                  "https://advancelocal-adapter-image-uploads.s3.amazonaws.com/image.al.com/home/bama-media/width600/img/news_impact/photo/burger-fijpg-57e7e5907630c2ad.jpg",
                  "https://img.thedailybeast.com/image/upload/c_crop,d_placeholder_euli9k,h_1439,w_2560,x_0,y_0/dpr_1.5/c_limit,w_1044/fl_lossy,q_auto/v1492718105/articles/2013/09/24/burger-king-s-new-french-fries-took-ten-years-to-develop/130923-gross-burger-tease_izz59e",
                  "https://aht.seriouseats.com/images/2012/02/20120221-193971-fast-food-fries-Burger-King-fries-2.jpg",
                ],
                "time": "15",
                "title": "Oatmeal Cookies",
              },
              "quantity": 1,
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        testID="checkoutList"
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 20,
                  "paddingVertical": 10,
                  "width": "100%",
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "fontWeight": "bold",
                    }
                  }
                >
                  Oatmeal Cookies
                </Text>
                <Text
                  style={
                    Object {
                      "color": "gray",
                      "fontSize": 12,
                    }
                  }
                >
                  Regular
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": 150,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "borderColor": "red",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "color": "red",
                      "paddingHorizontal": 30,
                      "paddingVertical": 5,
                    }
                  }
                >
                  1
                </Text>
                <Text>
                  $
                  20
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <RCTScrollView
        style={
          Object {
            "paddingHorizontal": 20,
          }
        }
      >
        <View>
          <View>
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "paddingTop": 30,
                }
              }
            >
              Bill Details
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "gray",
                  }
                }
              >
                Item Total
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                
                $
                20.00
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "gray",
                  }
                }
              >
                Delivery Fee
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                
                $
                5.00
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "gray",
                  }
                }
              >
                Free Ship
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                - 
                $
                5.00
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "gray",
                "borderBottomWidth": 1,
                "marginHorizontal": 30,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              To Pay
            </Text>
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              $
              20
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "gray",
                "borderBottomWidth": 1,
                "marginHorizontal": 30,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 10,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Enter discount code"
              rejectResponderTermination={true}
              style={
                Object {
                  "borderColor": "gray",
                  "borderRadius": 1,
                  "borderStyle": "dashed",
                  "borderWidth": 1,
                  "marginVertical": 10,
                  "paddingHorizontal": 20,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="applyBtn"
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 100,
                      "marginHorizontal": 10,
                    },
                    Object {
                      "backgroundColor": "gray",
                      "paddingHorizontal": 30,
                      "paddingVertical": 7,
                    },
                    Object {
                      "borderRadius": 10,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "fontWeight": "normal",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  APPLY
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "gray",
                "borderBottomWidth": 1,
                "marginHorizontal": 30,
              }
            }
          />
          <View
            style={
              Object {
                "paddingVertical": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Deliver To 
                Work
              </Text>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="changeAddressBtn"
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 100,
                        "marginHorizontal": 10,
                      },
                      Object {},
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 12,
                          "fontWeight": "normal",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    CHANGE
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "gray",
                  "paddingVertical": 10,
                }
              }
            >
              604 NT
            </Text>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="paymentBtn"
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 100,
                      "marginHorizontal": 10,
                    },
                    Object {
                      "backgroundColor": "red",
                      "paddingHorizontal": 60,
                      "paddingVertical": 10,
                    },
                    Object {
                      "borderRadius": 10,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "fontWeight": "normal",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  MAKE PAYMENT
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.5)",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderRadius": 20,
                "elevation": 5,
                "paddingVertical": 20,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
                "width": 700,
              }
            }
          >
            <RCTScrollView
              ItemSeparatorComponent={[Function]}
              ListFooterComponent={[Function]}
              ListHeaderComponent={[Function]}
              data={Array []}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              showsVerticalScrollIndicator={false}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View>
                <View
                  onLayout={[Function]}
                >
                  <Text>
                    Choose your address for delivery
                  </Text>
                </View>
                <View
                  onLayout={[Function]}
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="addAddressBtn"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderRadius": 100,
                            "marginHorizontal": 10,
                          },
                          Object {
                            "backgroundColor": "red",
                            "paddingHorizontal": 30,
                            "paddingVertical": 10,
                          },
                          Object {
                            "marginTop": 20,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 14,
                              "fontWeight": "normal",
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        Add New Address
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </Modal>
    </View>
  </View>
</View>
`;
